db/config.cc

```diff
@@ -983,8 +983,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , repair_multishard_reader_buffer_hint_size(this, "repair_multishard_reader_buffer_hint_size", liveness::LiveUpdate, value_status::Used, 1 * 1024 * 1024,
         "The buffer size to use for the buffer-hint feature of the multishard reader when running repair in mixed-shard clusters. This can help the performance of mixed-shard repair (including RBNO). Set to 0 to disable the hint feature altogether.")
     , repair_multishard_reader_enable_read_ahead(this, "repair_multishard_reader_enable_read_ahead", liveness::LiveUpdate, value_status::Used, false,
-        "The multishard reader has a read-ahead feature to improve latencies of range-scans. This feature can be detrimental when the multishard reader is used under repair, as is the case in repair in mixed-shard clusters."
-        " This know allows disabling this read-ahead (default), this can help the performance of mixed-shard repair (including RBNO).")
+        "The multishard reader has a read-ahead feature to improve latencies of range-scans. This feature can be detrimental when the multishard reader is used under repair, as is the case with repair in mixed-shard clusters."
+        " This configuration option allows disabling this read-ahead and may help the performance of repair in mixed-shard repair scenario (including RBNO).")
```

Comment:
The old text made it clear that this read-ahead is disabled by default. The new text says it "allows disabling" and makes it sound like it recommends the user to do it to solve some problems, but in fact, it looks like this is already the default.
Neither text - not before and not after - explains to me why this is even an option and when should a user consider setting it to "true" - if this small-table-scan option is detrimental for repair, why would a user set it to true? When should it be considered?

Reply:
We don't know when this option should be considered. It is here solely as a just-in-case kill-switch in case this optimization is found to go wrong in the future.